docs/workforce-management-framework/topics/ref-keycloak-securitystrategy.adoc

```diff
@@ -0,0 +1,40 @@
+[id='ref-keycloak-securitystrategy-{chapter}']
+= Defining the Security Strategy for {WFM-RC-NameShort} {WFM-RC-KeycloakName}
+
+This section includes:
+
+. xref:understanding-keycloak-session-management-{chapter}[Understanding {WFM-RC-KeycloakName} Session Management]
+. xref:defining-keycloak-access-control-{chapter}[Defining {WFM-RC-KeycloakName} Access Control]
+. xref:defining-keycloak-password-storage-{chapter}[Defining {WFM-RC-KeycloakName} Password Storage]
+. xref:keycloak-authentication-{chapter}[{WFM-RC-KeycloakName} Authentication]
+. xref:keycloak-authorization-{chapter}[{WFM-RC-KeycloakName} Authorization]
+
+[id='understanding-keycloak-session-management-{chapter}']
+== Understanding {WFM-RC-KeycloakName} Session Management
+The _express-session_ module provides session management functionality and
+the management of sessions is performed on the link:{WFM-RC-KeycloakURL}server_admin/topics/sessions/administering.html[*{WFM-RC-KeycloakName}* server]
+For more information on _express-session_, see link:https://github.com/expressjs/session[express-session].
+
+[id='defining-keycloak-access-control-{chapter}']
+== Defining {WFM-RC-KeycloakName} Access Control
+*{WFM-RC-KeycloakName}* provides role based access control and is configured on the *{WFM-RC-KeycloakName}* server.
+For more information about role based access control, see link:{WFM-RC-KeycloakURL}server_admin/topics/roles.html[*{WFM-RC-KeycloakName}* Roles].
+
+To see an example of how roles are protected, see the *{WFM-RC-KeycloakName}* link:./pro-Keycloak-implementaion.adoc[documentation].
+
+[id='defining-keycloak-password-storage-{chapter}']
+== Defining {WFM-RC-KeycloakName} Password Storage
+*{WFM-RC-KeycloakName}* uses the link:https://en.wikipedia.org/wiki/PBKDF2[BKDF2 algorithm] for password hashing by default.
+*{WFM-RC-KeycloakName}* also provides enforced link:{WFM-RC-KeycloakURL}server_admin/topics/authentication/password-policies.html[password policies] from the server.
+
+<ENG:Can you expand just a little on this? "from the server? - what does this mean?", "enforced passporot policies - does this mean one has to login using a password and username by default when using the Client App">
+
```

Comment:
http://www.keycloak.org/docs/2.1/server_admin_guide/topics/authentication/password-policies.html
Password policy is set of rules that all passwords must comply with - min length, special characters etc.